x-pack/plugins/apm/e2e/ingest-data/replay.js

```diff
@@ -70,34 +70,40 @@ function incrementSpinnerCount({ success }) {
 }
 let iterIndex = 0;
 
+function setItemMetaAndHeaders(item) {
+  const headers = {
+    'content-type': 'application/x-ndjson',
+  };
+
+  if (SECRET_TOKEN) {
+    headers.Authorization = `Bearer ${SECRET_TOKEN}`;
+  }
+
+  if (item.url === '/intake/v2/rum/events') {
+    if (iterIndex === userAgents.length) {
+      // set some event agent to opbean
+      setRumAgent(item);
+      iterIndex = 0;
+    }
+    headers['User-Agent'] = userAgents[iterIndex];
+    headers['X-Forwarded-For'] = userIps[iterIndex];
+    iterIndex++;
+  }
+  return headers;
+}
+
 function setRumAgent(item) {
-  item.body = item.body.replace(
-    '"name":"client"',
-    '"name":"opbean-client-rum"'
-  );
+  if (item.body) {
+    item.body = item.body.replace(
+      '"name":"client"',
+      '"name":"opbean-client-rum"'
+    );
```

Comment:
How risky is it to do a string replace? We might want to manipulate the body as an object rather than a string.